onedal/svm/svm.cpp

```diff
@@ -0,0 +1,227 @@
+                        .set_accuracy_threshold(params["accuracy_threshold"].cast<double>())
+                        .set_cache_size(params["cache_size"].cast<double>())
+                        .set_tau(params["tau"].cast<double>())
+                        .set_shrinking(params["shrinking"].cast<bool>());
+
+        if constexpr (is_cls || is_reg || is_nu_reg) {
+            desc.set_c(params["c"].cast<double>());
+        }
+        if constexpr (is_cls || is_nu_cls) {
+            desc.set_class_count(params["class_count"].cast<std::int64_t>());
+        }
+        if constexpr (is_reg) {
+            desc.set_epsilon(params["epsilon"].cast<double>());
+        }
+        if constexpr (is_nu_reg || is_nu_cls) {
+            desc.set_nu(params["nu"].cast<double>());
+        }
+
+        return desc;
+    }
+};
+
+template <typename Policy, typename Task>
+void init_train_ops(py::module_& m) {
+    m.def("train",
+          [](const Policy& policy,
+             const py::dict& params,
+             const table& data,
+             const table& responses,
+             const table& weights) {
+              using namespace svm;
+              using input_t = train_input<Task>;
+
+              train_ops ops(policy, input_t{ data, responses, weights }, params2desc{} );
+              return fptype2t { method2t { Task{}, kernel2t{ ops } } }(params);
+          });
+}
+
+template <typename Policy, typename Task>
+void init_infer_ops(py::module_& m) {
+    m.def("infer",
+          [](const Policy& policy,
+             const py::dict& params,
+             const svm::model<Task>& model,
+             const table& data) {
+              using namespace svm;
+              using input_t = infer_input<Task>;
+
+              infer_ops ops(policy, input_t{ model, data }, params2desc{} );
+              return fptype2t { method2t { Task{}, kernel2t{ ops } } }(params);
+          });
+}
+
+template <typename Task>
+void init_model(py::module_& m) {
+    using namespace svm;
+    using model_t = model<Task>;
+
+    auto cls =
+        py::class_<model_t>(m, "model")
+            .def(py::init())
+            .def(py::pickle(
+                [](const model_t& m) {
+                    return serialize(m);
+                },
+                [](const py::bytes& bytes) {
+                    return deserialize<model_t>(bytes);
+                }))
+            .def_property_readonly("support_vector_count", &model_t::get_support_vector_count)
+            .DEF_ONEDAL_PY_PROPERTY(support_vectors, model_t)
+            .DEF_ONEDAL_PY_PROPERTY(coeffs, model_t)
+            .DEF_ONEDAL_PY_PROPERTY(biases, model_t);
+
+    constexpr bool is_classification = std::is_same_v<Task, task::classification>;
+    constexpr bool is_nu_classification = std::is_same_v<Task, task::nu_classification>;
+
+    if constexpr (is_classification || is_nu_classification) {
+        // workaround for gcc which cannot deduce setters directly passed to def_property()
```

Comment:
for any version gcc?

Reply:
For latest gcc versions at least (9.1.0, 7.4.0)